pkg/server/restart.go

```diff
@@ -114,20 +115,26 @@ func (c *criService) recover(ctx context.Context) error {
 	// with best effort.
 
 	// Cleanup orphaned sandbox directories without corresponding containerd container.
-	if err := cleanupOrphanedSandboxDirs(sandboxes, filepath.Join(c.config.RootDir, "sandboxes")); err != nil {
+	if err := cleanupOrphanedSandboxDirs(sandboxes, filepath.Join(c.config.RootDir, sandboxesDir)); err != nil {
 		return errors.Wrap(err, "failed to cleanup orphaned sandbox directories")
 	}
+	if err := cleanupOrphanedSandboxDirs(sandboxes, filepath.Join(c.config.StateDir, sandboxesDir)); err != nil {
+		return errors.Wrap(err, "failed to cleanup orphaned volatile sandbox directories")
+	}
 
 	// Cleanup orphaned container directories without corresponding containerd container.
-	if err := cleanupOrphanedContainerDirs(containers, filepath.Join(c.config.RootDir, "containers")); err != nil {
+	if err := cleanupOrphanedContainerDirs(containers, filepath.Join(c.config.RootDir, containersDir)); err != nil {
 		return errors.Wrap(err, "failed to cleanup orphaned container directories")
 	}
+	if err := cleanupOrphanedContainerDirs(containers, filepath.Join(c.config.StateDir, containersDir)); err != nil {
+		return errors.Wrap(err, "failed to cleanup orphaned volatile container directories")
+	}
```

Comment:
Nit: all the `filepath.Join()` calls above should probably be moved to `helpers.go` and be resued by other functions to lower the chance of inconsistency. 
```go
func (c *criService) getContainerRootBaseDir() string {
    return filepath.Join(c.config.RootDir, containersDir)
}
````

Reply:
Will do.

Reply:
I found that there will be too many helper functions... :(
Since this is only used here, let's keep it `filepath.Join`. :)

Reply:
If you ever change the path, you'd have to track and change multiple files.
If you use `getContainerRootBaseDir()` in other calls, you'd only have to change it once.
E.g., `getContainerRooDir(id string) { return filepath.Join(getContainerRootBaseDir(), id) }`
Not that important, so up to you :-)